src/arch/xtensa/task.c

```diff
@@ -199,6 +203,8 @@ int arch_allocate_tasks(void)
 	/* irq high */
 	struct irq_task **high = task_irq_high_get();
 	*high = rzalloc(RZONE_SYS, SOF_MEM_CAPS_RAM, sizeof(**high));
+	if (!*high)
+		return -ENOMEM;
 
```

Comment:
This one is trickier here, because depending on the CONFIG_TASK_HAVE_PRIORITY_XXX you can end up with memory leaks.

But looking at the code it looks that it doesn't really matter if those allocation failed.

Reply:
I think it was already mentioned somewhere, that it looks like KW problem, because fail to alloc from system heap always ends in panic.